pjsip/src/pjsip/sip_uri.c

```diff
@@ -687,7 +688,7 @@ static pj_ssize_t other_uri_print(pjsip_uri_context_e context,
 
     /* Print scheme. */
     copy_advance(buf, uri->scheme);
-    *buf++ = ':';
+    copy_advance_char_check(buf, ':');
```

Comment:
rather OOT, but while on this area, perhaps this functions should also add NULL termination (and length minus one) too?